### PR TITLE
Remove addr and port reuse options for datagram sockets.

### PR DESCRIPTION
**Pull request by mpavelka**

### CODE REVIEW

**Review by ateska — CHANGES_REQUESTED** [review]
Ok. I came to conclusion that these flags needs to be configurable:

```
reuse: auto \ on \ off
```

Default is `auto` - meaning for TCP (stream) is on, otherwise off.

File: bspump/ipc/datagram.py [hunk 1]
```diff
@@ -123,8 +117,6 @@ def __init__(self, app, pipeline, id=None, config=None):
 
 			self.Socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
 			self.Socket.setblocking(False)
-			self.Socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-			self.Socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
```
Comment (ateska) [inline bspump/ipc/datagram.py:127]: `SO_REUSEADDR` is likely not applied to AF_UNIX at all ... I suggest to remove the setting from here.

**Comment by mpavelka** [comment]
Just to summarize:

**Unix Sockets** both stream and datagram will **not** have SO_REUSEADDR nor SO_REUSEPORT applied

**INET Sockets** both stream and datagram shall be configurable to apply SO_REUSEADDR and SO_REUSEPORT:
- For INET / Stream socket "auto" means **on**
- For INET / Datagram socket "auto" means **off**

@ateska, correct?

**Comment by ateska** [comment]
@mpavelka - correct.

**Pull request closed without merging**